src/common.speech/ServiceMessages/TurnStatusPayload.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+export interface ITurnStatusResponsePayload {
+    interactionId: string;
+    conversationId: string;
+    statusCode: any;
+}
+
+export class TurnStatusResponsePayload implements ITurnStatusResponsePayload {
+    private privMessageStatusResponse: ITurnStatusResponsePayload;
+
+    private constructor(json: string) {
+        this.privMessageStatusResponse = JSON.parse(json);
+    }
+
+    public static fromJSON(json: string): TurnStatusResponsePayload {
+        return new TurnStatusResponsePayload(json);
+    }
+
+    public get interactionId(): string {
+        return this.privMessageStatusResponse.interactionId;
+    }
+
+    public get conversationId(): string {
+        return this.privMessageStatusResponse.conversationId;
+    }
+
+    public get statusCode(): any {
+        // Payloads may contain a limited set of textual representations or a numeric status
+        // code. The textual values are here converted into numeric ones.
+        switch (this.privMessageStatusResponse.statusCode) {
```

Comment:
This is probably a better experience for the customer than "guess which random string the backend is using to indicate a time out," but it still feels odd returning a number to clearly indicate the status.

Reply:
The good news is that this is, for the time being, exclusively going to be standard HTTP codes, e.g. 200 for success, 400 for failure (Bad Request), 429 for Timeout (Bot Framework "too many requests" throttling), etc.. If there were a strongly-typed way to represent it (like "HttpStatusCode.OK"), I'd use it, but it seems like we'd have to define that ourselves--likely more harmful than helpful.